packages/gguf/src/types.ts

```diff
@@ -75,21 +86,43 @@ interface Tokenizer {
 	"tokenizer.ggml.bos_token_id": number;
 	"tokenizer.ggml.eos_token_id": number;
 	"tokenizer.ggml.add_bos_token": boolean;
-	"tokenizer.chat_template": string;
+	"tokenizer.chat_template"?: string;
 }
+interface NoTokenizer {
+	"tokenizer.ggml.model"?: undefined;
+}
+
+/// Models outside of llama.cpp: "rwkv" and "whisper"
 
-export type RWKV = ModelBase<"rwkv"> & { "rwkv.architecture_version": number };
-export type LLM = TransformerLLM | RWKV;
-export type Whisper = ModelBase<"encoder.whisper"> & ModelBase<"decoder.whisper">;
-export type Model = (LLM | Whisper) & Partial<Tokenizer>;
+export type RWKV = GGUFGeneralInfo<"rwkv"> &
+	ModelBase<"rwkv"> & {
+		"rwkv.architecture_version": number;
+	};
 
-export type GGUFMetadata = {
+// TODO: whisper.cpp doesn't yet support gguf. This maybe changed in the future.
+export type Whisper = GGUFGeneralInfo<"whisper"> &
+	ModelBase<"encoder.whisper"> &
+	ModelBase<"decoder.whisper"> & {
+		"whisper.encoder.mels_count": number;
+		"whisper.encoder.attention.head_count": number;
+		"whisper.decoder.attention.head_count": number;
+	};
+
+/// Types for parse output
+
+export enum GGUFType {
+	strict,
+	nonStrict,
```

Comment:
```suggestion
	STRICT,
	NON_STRICT,
```

stylitic nit. we usually prefer the enum values to be uppercase

Reply:
Thanks, that's done with https://github.com/huggingface/huggingface.js/pull/655/commits/5f547dd882ad07dad612225f6b8193845fb318a9